packages/x-tree-view/src/internals/plugins/useTreeViewItems/useTreeViewItems.tsx

```diff
@@ -231,24 +358,33 @@ export const useTreeViewItems: TreeViewPlugin<UseTreeViewItemsSignature> = ({
       getItemDOMElement,
       getItemTree,
       getItemOrderedChildrenIds,
+      getParentId,
     },
     instance: {
       getItemDOMElement,
       preventItemUpdates,
       areItemUpdatesPrevented,
+      addItems,
+      setTreeViewLoading,
+      setTreeViewError,
+      removeChildren,
     },
     contextValue: pluginContextValue,
   };
 };
 
 useTreeViewItems.getInitialState = (params) => ({
-  items: updateItemsState({
-    disabledItemsFocusable: params.disabledItemsFocusable,
-    items: params.items,
-    isItemDisabled: params.isItemDisabled,
-    getItemId: params.getItemId,
-    getItemLabel: params.getItemLabel,
-  }),
+  items: {
+    ...updateItemsState({
+      disabledItemsFocusable: params.disabledItemsFocusable,
+      items: params.items,
+      isItemDisabled: params.isItemDisabled,
+      getItemId: params.getItemId,
+      getItemLabel: params.getItemLabel,
+    }),
+    loading: false,
```

Comment:
Can we move `loading` and `error` inside `updateItemsState` so that it handles the full state?

Reply:
Hmm, I'm not 100% we should. The initial state is the only place where we set both the loading and error state and call `updateItemsState` together. We call `updateItemsState` every time we load items on expansion. The loading and error state of the entire Tree View changes only once, when we do the initial load. So basically, `updateItemsState` doesn't really need to care about those I think 🤔

Reply:
Hmm my issue is that `updateItemsState` only updates some of the state.
But I guess we can clean that up later.
We could imagine just renaming the function to something like `updateItemsStateLookups` which is what it does right?

Reply:
It's what I was going to suggest [here](https://github.com/mui/mui-x/pull/15308#discussion_r1950733995). Technically, this method doesn't really update the state - it processes an array of items that is later added to the state in the shape of lookups. So maybe renaming it to `processItemsLookups` helps?

Reply:
Would be even better :ok_hand: